File: app/src/main/java/com/yellowtwigs/knockin/controller/SwipeToDeleteCallback.java
package com.yellowtwigs.knockin.controller;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

import com.yellowtwigs.knockin.ui.adapters.NotifPopupRecyclerViewAdapter;

public class SwipeToDeleteCallback extends ItemTouchHelper.SimpleCallback {
    private NotifPopupRecyclerViewAdapter mAdapter;

    public SwipeToDeleteCallback(NotifPopupRecyclerViewAdapter adapter) {
        super(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT);
        mAdapter = adapter;
    }

    @Override
    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
        return false;
    }

    @Override
    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
        int position = viewHolder.getAdapterPosition();
        mAdapter.deleteItem(position);
    }
}

File: app/src/main/java/com/yellowtwigs/knockin/ui/notifications/SwipeToDeleteCallback.kt
package com.yellowtwigs.knockin.ui.notifications

import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

class SwipeToDeleteCallback(
//    private val mAdapter: NotifPopupRecyclerViewAdapter?
) : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT) {
    override fun onMove(
        recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder,
        target: RecyclerView.ViewHolder
    ): Boolean {
        return false
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
        val position = viewHolder.adapterPosition
//        mAdapter?.deleteItem(position)
    }
}